pkg/utils/nodeutils.go

```diff
@@ -84,7 +84,8 @@ func IsMcfgPoolUsingKC(pool *mcfgv1.MachineConfigPool) (bool, string, error) {
 	var currentKCMC string
 	for i := range pool.Spec.Configuration.Source {
 		kcName := pool.Spec.Configuration.Source[i].Name
-		if strings.Contains(kcName, generatedKubelet) {
+		// The prefix has to start with 99 since the kubeletconfig generated machine config will always start with 99
```

Comment:
Question: Is there no label or annotation to use instead?

Reply:
No, unfortunately I don't think there are annotation/label we can use to identify if mc is generated by kubeletconfig object, but we can use owner GetOwnerReferences to find out if it is owned by Kubeletconfig object 

```
[vincent@fedora-ws compliance-operator]$ oc get mc 99-master-generated-kubelet -o yaml
apiVersion: machineconfiguration.openshift.io/v1
kind: MachineConfig
metadata:
  annotations:
    machineconfiguration.openshift.io/generated-by-controller-version: 5da6b0129a14cb44b9d7218235992933180546bf
  creationTimestamp: "2022-01-17T17:49:25Z"
  generation: 1
  labels:
    machineconfiguration.openshift.io/role: master
  name: 99-master-generated-kubelet
  ownerReferences:
  - apiVersion: machineconfiguration.openshift.io/v1
    blockOwnerDeletion: true
    controller: true
    kind: KubeletConfig
    name: kubelet-config-test-master
    uid: 4377ed1e-e71a-4d5c-83ef-324bd3a302d7
  resourceVersion: "135039"
  uid: da08d8ab-8d48-4eb5-bcc6-146b7c52661a
spec:
```

Reply:
I updated the code to check if mc is created by Kubeletconfig in complianceremediation_controller.go

Reply:
thanks, this works for me